### PR TITLE
Port Group E test_transforms.py to pytest

### PR DESCRIPTION
**Pull request by artifiitr**
To port Group E in test_transforms.py to pytest as discussed in #3945

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -1315,6 +1196,130 @@ def test_random_erasing(self):
         t.__repr__()
 
 
+def test_convert_image_dtype_float_to_float(self):
+    for input_dtype, output_dtypes in cycle_over(float_dtypes()):
+        input_image = torch.tensor((0.0, 1.0), dtype=input_dtype)
+        for output_dtype in output_dtypes:
```
Comment (NicolasHug) [inline test/test_transforms.py:1202]: Thanks for the PR @artifiitr !

We should remove `self` from the function and also try to parametrize over `input_dtype` and `output_dtype`. This way we should be able to get rid of the self.subTest as well. You can find a complete example of parametrization here https://github.com/pytorch/vision/pull/3907/files.

**Comment by NicolasHug** [comment]
Hi @artifiitr , thanks again for the PR!
Are you still interested in working on it?

File: test/test_transforms.py (hunk 1 above)
Comment (artifiitr) [reply test/test_transforms.py:1202]: @NicolasHug , In 1200th line, output_dtype**s** contains only one value acccording to cycle_over(float_dtypes), So can I eliminate output_dtype**s** from above function and use only output_dtype?

Comment (NicolasHug) [reply test/test_transforms.py:1202]: From what I can tell this yields 2 elements, not just one:

```
(torch.float32, (torch.float64,))
(torch.float64, (torch.float32,))
```

let's just change this into a parametrization like

```py
@pytest.mark.parametrize('input_dtype, output_dtypes', cycle_over(float_dtypes()))
def test_blahblah(input_dtype, output_dtypes):
```

Comment (artifiitr) [reply test/test_transforms.py:1202]: So shall I keep the for loop in 1202 line?

Comment (NicolasHug) [reply test/test_transforms.py:1202]: no, the for loops should be repalced with parametrization. You can refer to the porting guide I wrote in https://github.com/pytorch/vision/issues/3945#issue-909493806, there are some illustration links

Comment (sahilg06) [reply test/test_transforms.py:1202]: @Nicholas, I think he is trying to say that
```
for output_dtype in output_dtypes:
```
in the above line , output_dtypes is a list of only 1 item. So aren't output_dtype and output_dtypes same in above function? And we can parametrize one out of them only.

Comment (artifiitr) [reply test/test_transforms.py:1202]: @NicolasHug Outer loop can be parametrized as u said. But inner loop is dependent on outer one. In the porting guide, I couldn't find parameterization of such loop. So, how should I parametrize it?

Comment (NicolasHug) [reply test/test_transforms.py:1202]: Sorry for the confusion @artifiitr , indeed here we can parametrize over  

```py
@pytest.mark.parametrize('input_dtype, output_dtypes', cycle_over(float_dtypes()))
def test_blahblah(input_dtype, output_dtypes):
```
but it seems that we'll need to keep the inner loop:

```py
for output_dtype in output_dtypes:
    ...
```

**Comment by artifiitr** [comment]
@NicolasHug , I had to make changes in environment, so have created a new pull request. The link to new PR is: https://github.com/pytorch/vision/pull/4026

**Pull request closed without merging**